packages/acceptance-tests/pkg-tests-specs/sources/commands/npm/login.test.ts

```diff
@@ -211,5 +211,35 @@ describe(`Commands`, () => {
         expect({code, stdout, stderr}).toMatchSnapshot();
       }),
     );
+
+    test(
+      `it should store npmAlwaysAuth when passed as option`,
+      makeTemporaryEnv({}, async ({path, run, source}) => {
+        const rcPath = ppath.join(path, PortablePath.parent, SPEC_RC_FILENAME);
+        await xfs.writeFilePromise(rcPath, ``);
+
+        let code;
+        let stdout;
+        let stderr;
+
+        try {
+          ({code, stdout, stderr} = await run(`npm`, `login`, `--always-auth`, {
+            env: {
+              YARN_INJECT_NPM_USER: validLogins.fooUser.username,
+              YARN_INJECT_NPM_PASSWORD: validLogins.fooUser.password,
+              YARN_RC_FILENAME: SPEC_RC_FILENAME,
+            },
+          }));
+        } catch (error) {
+          ({code, stdout, stderr} = error);
+        }
+
+        const finalRcFileContent = await xfs.readFilePromise(ppath.join(path, PortablePath.parent, SPEC_RC_FILENAME), `utf8`);
+        const cleanFileContent = cleanupFileContent(finalRcFileContent);
+
+        expect(cleanFileContent).toMatchSnapshot();
+        expect({code, stdout, stderr}).toMatchSnapshot();
```

Comment:
Not fan of using snapshots for what should be a static check - can you rather use the pattern found in `packages/acceptance-tests/pkg-tests-specs/sources/commands/config/get.test.js`:

```ts
await expect(run(`config`, `get`, `npmAuthToken`, `--no-redacted`)).resolves.toMatchObject({
  stdout: `foobar\n`,
});
```

Reply:
I had to work around the fact that the registry is dynamic, so it looks slightly different from the code you pointed to. I didn't want to make the test dependent on the other properties in npm registry configuration.